ide/html.editor/src/org/netbeans/modules/html/editor/hints/other/AddMissingAltAttribute/AddMissingAltAttributeHint.java

```diff
@@ -0,0 +1,89 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.netbeans.modules.html.editor.hints.other.AddMissingAltAttribute;
+
+import java.awt.EventQueue;
+import java.util.Collections;
+import javax.swing.text.BadLocationException;
+import org.netbeans.modules.csl.api.Hint;
+import org.netbeans.modules.csl.api.HintFix;
+import org.netbeans.modules.csl.api.OffsetRange;
+import org.netbeans.modules.csl.api.RuleContext;
+import org.netbeans.modules.html.editor.utils.HtmlTagContextUtils;
+import org.openide.util.Exceptions;
+
+/**
+ *
+ * @author Chris
+ */
+public class AddMissingAltAttributeHint extends Hint {
+
+    public AddMissingAltAttributeHint(RuleContext context, OffsetRange range) {
+        super(AddMissingAltAttributeRule.SINGLETON,
+                AddMissingAltAttributeRule.SINGLETON.getDisplayName(),
+                context.parserResult.getSnapshot().getSource().getFileObject(),
+                range,
+                Collections.<HintFix>singletonList(new AddMissingAltAttributeHintFix(context)),
+                10);
+    }
+
+    private static class AddMissingAltAttributeHintFix implements HintFix {
+
+        RuleContext context;
+
+        public AddMissingAltAttributeHintFix(RuleContext context) {
+            this.context = context;
+        }
+
+        @Override
+        public String getDescription() {
+            return AddMissingAltAttributeRule.SINGLETON.getDisplayName();
+        }
+
+        @Override
+        public void implement() throws Exception {
+            EventQueue.invokeLater(new Runnable() {
+
+                @Override
+                public void run() {
```

Comment:
feel free to bump the language level to 1.8 so that you can use lambdas.

Reply:
did it